### ISSUE TITLE
Warning messages: closing unused connections

### ISSUE THREAD

**Issue by irenetorrubia** [rdotsch/rcicr#50]
It's me again. The code did run*, but returned the following warning:

> There were 12 warnings (use warnings() to see them)
> > warnings()
> Warning messages:
> 1: closing unused connection 14 (<-PC:11588)
> 2: closing unused connection 13 (<-PC:11588)
> 3: closing unused connection 12 (<-PC:11588)
> 4: closing unused connection 11 (<-PC:11588)
> 5: closing unused connection 10 (<-PC:11588)
> 6: closing unused connection 9 (<-PC:11588)
> 7: closing unused connection 8 (<-PC:11588)
> 8: closing unused connection 7 (<-PC:11588)
> 9: closing unused connection 6 (<-PC:11588)
> 10: closing unused connection 5 (<-PC:11588)
> 11: closing unused connection 4 (<-PC:11588)
> 12: closing unused connection 3 (<-PC:11588)
> There were 12 warnings (use warnings() to see them)

I assume PC:11588 is my computer. I tried to give a quick look to the code to see if there was a syntax element that could lead to this error, but maybe it is called from another function. 

Best, 

Irene. 

*stimuli <- generateStimuli2IFC(base_face_files = list(refimage="RefImage.jpg"), n_trials = 10, img_size =  512, stimulus_path = "./stimuli", label = "rcic", use_same_parameters = TRUE, seed = 1, maximize_baseimage_contrast = F, noise_type = "sinusoid", nscales = 5, sigma = 25, returnAsList = TRUE)

**Comment by rdotsch** [comment]
This seems to be tied to the parallel processing. Did you get the correct output? In that case I think it's safe to ignore the warnings. Thanks for reporting and keep 'em coming!

**Comment by aucuparia** [comment]
@rdotsch Correct. I think we would want to hide these messages so as not to confuse users, but it's a low priority issue.

**Comment by irenetorrubia** [comment]
@rdotsch my apologies, I missed the message. It did yield the correct output.

**Comment by rdotsch** [comment]
These messages only occur if in an earlier run during the same session R was interrupted before it could terminate the parallel processes (which then keep on running). We should catch these kinds of interruptions and kill the parallel processes.

**Comment by rdotsch** [comment]
**`closing unused connections` warnings.**

Fixed. Clusters are now stopped via `on.exit()`, so workers are released even when an error interrupts the loop rather than being orphaned.

Related improvement: `ncores = 1` no longer creates a cluster at all. It previously started a one-worker cluster to run what is a sequential loop, which is where a lot of these warnings came from.

This is fixed in `main`.

The package has just been through a substantial modernization pass: all seven known correctness bugs are fixed, there is now a test suite and CI, and every fix is covered by a regression test. A golden-master test pins the numeric output of the default pipeline to what it produced *before* the fixes, so default-configuration results are unchanged — that claim is backed by a test you can run, not by assertion.

```r
install.packages("remotes")
remotes::install_github("rdotsch/rcicr")
```

`install.packages("rcicr")` still does not work: the package was archived from CRAN in 2021 because the maintainer email address had gone stale. That address is fixed and a return to CRAN is being prepared, but GitHub is the source for now.

If you have results computed with an older version, please read the **"Reproducibility impact"** section of [`NEWS.md`](https://github.com/rdotsch/rcicr/blob/main/NEWS.md) before re-running — most fixes turn a crash into working behaviour and cannot change a number you already have, but two of them genuinely do change infoVal, and the conditions are spelled out there.

Thanks for reporting it, and apologies it sat for so long.
